provider/providertest/integration.go

```diff
@@ -0,0 +1,394 @@
+package providertest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/cloudquery/cq-provider-sdk/cqproto"
+	"github.com/cloudquery/cq-provider-sdk/logging"
+	"github.com/cloudquery/cq-provider-sdk/provider"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+	"github.com/go-test/deep"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/hashicorp/terraform-exec/tfexec"
+	"github.com/jackc/pgx/v4"
+	"github.com/tmccombs/hcl2json/convert"
+)
+
+const (
+	tfDir    = "./int_test/tf/"
+	tfOrigin = "./testData/"
+)
+
+type ResourceIntegrationTestData struct {
+	Table               *schema.Table
+	Config              interface{}
+	Resources           []string
+	Configure           func(logger hclog.Logger, data interface{}) (schema.ClientMeta, error)
+	Suffix              string
+	Prefix              string
+	VerificationBuilder func(res *ResourceIntegrationTestData) ResourceIntegrationVerification
+}
+
+// ResourceIntegrationVerification - a set of verification rules to query and check test related data
+type ResourceIntegrationVerification struct {
+	Name           string
+	ForeignKeyName string
+	ExpectedValues []ExpectedValue
+	Filter         func(sq squirrel.SelectBuilder, res *ResourceIntegrationTestData) squirrel.SelectBuilder
+	Relations      []*ResourceIntegrationVerification
+}
+
+// ExpectedValue - describes the data that expected to be in database after fetch
+type ExpectedValue struct {
+	Count int                    // expected count of items
+	Data  map[string]interface{} // expected data of items
+}
+
+// IntegrationTest - creates resources using terraform, fetches them to db and compares with expected values
+func IntegrationTest(t *testing.T, providerCreator func() *provider.Provider, resource ResourceIntegrationTestData) {
```

Comment:
This function is quite big maybe split it up into fewer parts -> setup -> deploy -> test -> teardown